### PR TITLE
Add @types/yup.

### PR DESCRIPTION
**Pull request by squidsoup**
## Done

- Add types for `yup`.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- n/a

## Fixes

Fixes: n/a

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

File: ui/src/app/kvm/views/KVMList/AddKVMForm/AddKVMForm.tsx [hunk 1]
```diff
@@ -28,7 +28,9 @@ import type { PowerType } from "app/store/general/types";
 import type { TSFixMe } from "app/base/types";
 import zoneSelectors from "app/store/zone/selectors";
 
-const generateAddKVMSchema = (parametersSchema: TSFixMe) =>
+const generateAddKVMSchema = (
+  parametersSchema: Yup.ObjectSchemaDefinition<TSFixMe>
```
Comment (squidsoup) [inline ui/src/app/kvm/views/KVMList/AddKVMForm/AddKVMForm.tsx:32]: As this is a dynamic schema, I'm not sure how to type it...

File: ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.tsx [hunk 2]
```diff
@@ -154,7 +154,8 @@ const ComposeForm = ({ setSelectedAction }: Props): JSX.Element | null => {
 
     const ComposeFormSchema = Yup.object().shape({
       architecture: Yup.string(),
-      cores: Yup.number("Cores must be a positive number.")
+      cores: Yup.number()
+        .positive("Cores must be a positive number.")
```
Comment (squidsoup) [inline ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.tsx:158]: yay, typescript finds bugs

File: ui/src/app/kvm/views/KVMList/AddKVMForm/AddKVMForm.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/kvm/views/KVMList/AddKVMForm/AddKVMForm.tsx:32]: Yeah I can't see how this is supposed to be done either.

File: ui/.betterer.results [hunk 3]
```diff
@@ -30,7 +30,7 @@ exports[`stricter compilation`] = {
...
     ],
     "src/app/base/components/Slider/Slider.tsx:753275532": [
-      [0, 18, 51, "Could not find a declaration file for module \'@canonical/react-components/dist/components/Field\'. \'/home/multipass/code/maas-ui/node_modules/@canonical/react-components/dist/components/Field/index.js\' implicitly has an \'any\' type.\\n  Try \`npm install @types/canonical__react-components\` if it exists or add a new declaration (.d.ts) file containing \`declare module \'@canonical/react-components/dist/components/Field\';\`", "1535046059"]
+      [0, 18, 51, "Could not find a declaration file for module \'@canonical/react-components/dist/components/Field\'. \'/Users/kit/src/canonical/local/maas-ui/node_modules/@canonical/react-components/dist/components/Field/index.js\' implicitly has an \'any\' type.\\n  Try \`npm install @types/canonical__react-components\` if it exists or add a new declaration (.d.ts) file containing \`declare module \'@canonical/react-components/dist/components/Field\';\`", "1535046059"]
```
Comment (huwshimi) [inline ui/.betterer.results:33]: Huh, I wonder if we're doing something wrong...

**Pull request merged**